### ISSUE TITLE
Metanorma templates dir may have not permissions to write

### ISSUE THREAD

**Issue by CAMOBAP795** [metanorma/metanorma-cli#53]
On OSX to install gems you need to have admin/root access

As a result when I tried to run `metanorma new test --type iso --doctype standard` (in the scope of https://github.com/metanorma/mn-templates-iso/issues/1 task) I got the error:

<details><summary>Strack trace</summary>
<p>

```bash
metanorma new --type iso --doctype standard iso-rice/document.adoc
You've an existing document with the /Users/camobap/Developers/Projects/riboseinc/mn-templates-iso/iso-rice/document.adoc
Still want to continue, and overwrite the existing one? (yes/no): yes
/Library/Ruby/Gems/2.3.0/gems/git-1.5.0/lib/git/lib.rb:967:in `command': git  clone '--' 'https://github.com/metanorma/mn-templates-iso' '/Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/templates/iso'  2>&1:fatal: could not create work tree dir '/Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/templates/iso': Permission denied (Git::GitExecuteError)
	from /Library/Ruby/Gems/2.3.0/gems/git-1.5.0/lib/git/lib.rb:77:in `clone'
	from /Library/Ruby/Gems/2.3.0/gems/git-1.5.0/lib/git/base.rb:29:in `clone'
	from /Library/Ruby/Gems/2.3.0/gems/git-1.5.0/lib/git.rb:97:in `clone'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/git_template.rb:78:in `clone_git_template'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/git_template.rb:67:in `download_template'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/git_template.rb:26:in `find_or_download'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/git_template.rb:36:in `find_or_download_by'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/generator.rb:61:in `find_template'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/generator.rb:74:in `type_specific_template'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/generator.rb:47:in `create_metanorma_document'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/generator.rb:21:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/generator.rb:35:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/command.rb:65:in `create_new_document'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli/command.rb:16:in `new'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /Library/Ruby/Gems/2.3.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/lib/metanorma/cli.rb:56:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/metanorma-cli-1.1.7/exe/metanorma:14:in `<top (required)>'
	from /usr/local/bin/metanorma:22:in `load'
	from /usr/local/bin/metanorma:22:in `<main>'
```

</p>
</details>

Should we handle this somehow? I hope we should use `$HOME/.metanorma` or something like this to keep templates

cc @ronaldtse @opoudjis

**Comment by ronaldtse** [comment]
This is strange, I'm not getting this. This is not trying to create a new gem.

It seems that the command is trying to clone the template repository into your Ruby library path? Any idea @abunashir ?

**Comment by CAMOBAP795** [comment]
@ronaldtse this is recent functionality AFAIK in metanorma https://github.com/metanorma/metanorma-cli/commit/fe0faaae11170b56b24c116146aa3d6f67e55216

**Comment by ronaldtse** [comment]
Yes, but this is released functionality (gen version 1.1.7) and supposed to be stable. We need to fix this if the git clone is using system paths. It must use somewhere in the homedir.

**Comment by abunashir** [comment]
@ronaldtse Yes, we were using the gem directory to store our templates which might create some permission issue if we don't have the proper permission. 

Like I mentioned in the PR, this is really a good idea to use the home directory, so if we want we can also use the same cache version throughout multiple versions. 

In addition, we should also check if the template directory is writable or not, and if not then raise an error with proper messages. @CAMOBAP795 Do you mind taking care of this one?

**Comment by CAMOBAP795** [comment]
@abunashir good catch, I will care that case

**Comment by ronaldtse** [comment]
@abunashir is this now resolved with #64 ?

**Comment by abunashir** [comment]
Yap, closing it now :)
